packages/buffetjs-core/src/components/DatePicker/index.js

```diff
@@ -35,107 +51,105 @@ class DatePicker extends React.PureComponent {
+      displayedDate: target.value,
+      isFocused: false,
+    }));
+
+    timer = setTimeout(() => {
+      handleDateChange(moment(target.value, 'MM/DD/YYYY'));
+    }, wait);
+  };
 
+  const handleDateChange = date => {
     if (moment(date).isValid()) {
-      this.setState(
-        { date, displayedDate: date.format('MM/DD/YYYY'), visible: true },
-        () => onChange({ target: { name, type: 'date', value: date } })
-      );
+      onChange({ target: { name, type: 'date', value: date } });
+
+      setState(prevState => ({
+        ...prevState,
+        date,
+        displayedDate: date.format('MM/DD/YYYY'),
+      }));
     }
   };
 
-  handleFocusChange = ({ focused }) => {
-    /* istanbul ignore next */
-    this.setState({ isFocused: focused });
+  const handleDateClick = date => {
+    handleDateChange(date);
+
+    toggleDatepicker(false);
   };
 
-  handleOutsideClick = () => {
-    this.setState({
-      visible: false,
-    });
+  const handleOutsideClick = () => {
+    toggleDatepicker(false);
   };
 
-  handleTabClick = ({ keyCode, which }) => {
+  const handleTabClick = ({ keyCode, which }) => {
     const code = keyCode || which;
 
     if (code === 9) {
-      this.handleOutsideClick();
+      handleOutsideClick();
     }
   };
 
-  showDatepicker = () => {
-    this.setState({
-      visible: true,
-    });
+  const toggleDatepicker = shown => {
+    setState(prevState => ({
+      ...prevState,
+      visible: shown,
+      isFocused: shown,
+    }));
   };
 
-  render() {
-    const {
-      className,
-      disabled,
-      displayFormat,
-      id,
-      name,
-      readOnly,
-      tabIndex,
-    } = this.props;
-    const { date, displayedDate, isFocused, visible } = this.state;
-    let dateValue = date ? moment(date).format(displayFormat) : '';
-
-    if (visible) {
-      dateValue = displayedDate;
-    }
-
-    return (
-      <StyledDatepicker className={`${className} ${visible && 'is-open'}`}>
-        <div>
-          <Input
-            disabled={disabled}
-            type="text"
-            name="start_date"
-            id={id || name}
-            value={dateValue}
-            readOnly={readOnly}
-            onChange={this.handleChange}
-            icon={<FontAwesomeIcon icon={faCalendarAlt} />}
-            onClick={this.showDatepicker}
-            onFocus={this.showDatepicker}
-            onKeyDown={this.handleTabClick}
-            tabIndex={tabIndex}
-          />
-        </div>
-        {visible && (
-          <DayPickerSingleDateController
-            date={date}
-            focused={isFocused}
-            numberOfMonths={1}
-            keepOpenOnDateSelect
-            onFocusChange={this.handleFocusChange}
-            onDateChange={this.handleDateChange}
-            onOutsideClick={this.handleOutsideClick}
-          />
-        )}
-      </StyledDatepicker>
-    );
-  }
+  return (
+    <StyledDatepicker className={`${className} ${state.visible && 'is-open'}`}>
```

Comment:
Why `state.visible` is not passed as props here? As it is a styled-component, this behavior should be handled by props.